### ISSUE TITLE
Horrible UX and setup bugs – a first-time user experiency story

### ISSUE THREAD

**Issue by fbruetting** [futo-org/grayjay-android#782]
Hello, I usually create more precise issues to help developers, but in this case I honestly don’t know where to start. First of all, your project idea is really great! So thanks for that! But you throw a software onto the market, which even for IT people is overly complicated and also too buggy – it’s far to technical for normal people to get around with (I’m pretty sure most people quit at setup time, before even playing around with the app itself). To name some examples here’s my experience after skipping the YouTube tutorials (to be precise: I skipped them a week before, at installation time, but after one week trying to play around with the app, I forgot that tutorials existed at all. Still, apps should be self-explanatory):

* First, I wanted to look into the settings, to see what’s going on there and what I can tune. But I was presented a list of (felt) a hundred settings of totally different categories in one gigantic row, of which most were pretty irrelevant at first. No grouping, no selection via hidden settings / third level menues etc. So I immediately closed the settings, because I didn’t want to consume that whole list for getting a grasp of the options I might being able to activate/modify.

* Then I saw, that I can import my NewPipe list. Awesome! So I tried.
  * But first, parsing my list of 65 URLs took about 20 seconds like I wondered if that app mines half a Bitcoin…?
  * Then I seleted all channels and pressed “import”. But instead of one check for “OK, all URL are YouTube URLs, YouTube is not installed, let’s present one according dialog that the YouTube source needs to be present”, I was presented with 65 error messages showing up one after another that every single URL cannot be fetched. It took about one minute until the presentation was over.
    * Alternate solution: Activate the YouTube source (or all sources) by default.
    * Btw: With no active sources, it’s a good idea for the start screen to show for example “You first need to install and activate a source. Go to the sources menu or click HERE to do so.”
  * So I found out, I had to add appropriate sources first – but why wasn’t there an error message at importing, or (way better) a dialog like “The Source *YouTube* is necessary for 65/65 channels of yor NewPipe list. Do you want to activate this Source?” for doing so in-place.
  * I added YouTube, tried it again, but still an error. Then I recognized that I also had to activate if after installation. 🙄️
  * I activated YouTube and tried to import my list again. That process took over two minutes, so I killed GrayJay. At reopening the application, I recognized that just the first channel has been imported. I wanted to import the list again, but the “Import” button was gone from the subscriptions menu…?! There also was no “+” button in the top bar which I expected. As I wasn’t able to find that import capability anywhere in the app, I deleted that first channel to get an empty list with the import-button visible again, to import my list. That then worked out. In contrast to the first attempt (before activating the YouTube source) that import of 65 channels now happend instantly – bravo.

That is my story up to now. As it was pretty cumbersome to import 65 YouTube channels, I then felt to write that issue here. Non-technical people would have given up instantly, at least the vast majority of them. The whole process up to this point was reproducible, including stalling import etc. after deleting app-data.

The idea of your app is great and I love to get it being successful. But the way it currently works prevents that. It would be great if mentioned bugs would get fixed, hints be displayed and the UI getting cleaned up (especially dividing settings into menus, putting the “sources” menu (i.e. bottom-bar button) into settings menu, putting all fine-grain-settings in even deeplier nested menus and so on).

I had a similar experience with Matrix/Riot/Element. We had a loose group of 50 people in a voluntary organization who were open to try Riot/Element two years ago. But because of UX issues and bugs, more and more of them were turned off and wanted everybody to switch to WhatsApp (so they rather gave their phone number to everyone than to deal with UX issues). In the end, even after I tried to help out with their bugs they blamed „stupid OSS apps for being crap” which was pretty counter-productive for everybody involved.

So to sum it up: I’m very convinced that getting the UX simplified helps adoption WAY more than adding feature A, crpyto feature B, privacy feature C and so an. I had no time to check out your Circles app – but even when adoption does’t matter here in case of GrayJay, if Circles behaves similarly, it’s a big dealbreaker, because that heavily depends on adoption rate.

**Comment by koen-futo** [comment]
Thank you for your feedback. With regards to why it takes so long to parse 65 URLs, it is because we actually need to request each channel that needs to be subscribed to in order to make the data-flow work out. With regards to your other points,  we will consider each and see how it can be improved.

**Comment by TerryMartinez** [comment]
Funny enough i imported 300+channel from YT directly to grayjay from grayjay features (2024 grayjay)
